enhancements/node-tuning/secure-shortname-image-pulls.md

```diff
@@ -0,0 +1,216 @@
+in the alias table.
+
+## Motivation
+
+We are always trying to make OpenShift more secure. While we do highly discourage the usage of
+shortnames in favour of fully qualified names, there are users who still prefer using shortnames
+so we have updated the pull logic to be more secure than just checking a list of registries and pulling
+from the first location where a match is found.
+
+So, to make the shortname use-case more secure, the c/image library was updated to check an
+alias table when available. An alias table is a list of image names with shortnames pointing to their
+corresponding fully qualified image names. We can use this new feature to ensure that any Red Hat and OpenShift
+images that are referred to by short name will always resolve to our official locations, hence
+guaranteeing that those images will be official.
+
+### Goals
+
+- Secure where the official Red Hat and OpenShift images are pulled from when using shortnames
+- The alias table list will be maintained by us and be specific to the OpenShift use-case
+- The MCO will be used to lay down the alias table
+
+### Non-Goals
+
+- This is not encouraging the use of shortnames. It is instead focused on making the few use-cases
+  that are out there more secure
+
+## Proposal
+
+Enable the shortname alias logic in the cri-o code so that when an image is being pulled and
+referred to by shortname, cri-o will check the alias table first and if there is a match it
+will pull the image from that location. If there is no match it will fallback to check the
+`unqualified-search-registries` list in order and pull the image from the first location where
+a match is found - note, this is what is done currently.
+
+We will maintain the alias table shipped on the OpenShift node via the MCO. We will create an alias
+table for the official Red Hat and OCP images based on a subset of the aliases in the existing tables
+that are shipped in RHEL. We are doing this as the tables shipped with RHEL have a lot of images from
+the community which are not required in the OpenShift use-case.
+
+Note: This proposal is a security enhancement around the shortnames use case and is not an API change.
+
+### User Stories
+
+#### As a user, I would like to use image short names when running my workloads
+
+When the user uses shortname for official Red Hat and OpenShift images that have a matching alias in the
+alias table, the image will be pulled from the matching location specified there. This will ensure
+the security of where these images are being pulled from reducing the risk of spoofing.
+
+### API Extensions
+
+No API changes are needed for this proposal.
+
+### Implementation Details/Notes/Constraints [optional]
+
+Implementing this enhancement requires changes in:
+
+- cri-o/cri-o
+- openshift/machine-config-operator
+
+Update the cri-o code to enable the shortname alias logic to `permissive` mode. This will ensure
```

Comment:
This seems to refer to the code impacted by https://github.com/cri-o/cri-o/pull/4983/files , and is incorrect about what that option does.

Aliases are _already_ interpreted; compare https://github.com/containers/image/issues/1448#issuecomment-1018501551 and see https://bugzilla.redhat.com/show_bug.cgi?id=2021026 for evidence.

Reply:
So looks like there is truly no difference between the `permissive` and `disabled` mode when it comes to alias look up? 
The difference seems to be in whether the alias is recorded in the table when a match via `unqualified-search-registries` happen - is that correct?
From what I understood, `permissive` meant look at alias table, if no match is found don't prompt the user asking where to pull from, just check all the registries under `unqualified-search-registries`. And `disabled` meant don't look at any alias table, just check all the registries under `unqualified-search-registries`. This is why we changed cri-o to disabled mode when the first bug occurred https://bugzilla.redhat.com/show_bug.cgi?id=1969599.

If there is truly no difference, then would it matter whether it is in `permissive` or `disabled` mode when it comes to alias look up? We definitely don't want to do `enforcing` mode as that would throw an error when it can't prompt the user to select a location to pull from when an alias is not found.

Reply:
> So looks like there is truly no difference between the `permissive` and `disabled` mode when it comes to alias look up?

See https://github.com/containers/image/blob/70982d037a7a006fd3806dfb0882840aac2e2259/pkg/shortnames/shortnames.go#L302 , the mode value is not consulted at all at that point.

> The difference seems to be in whether the alias is recorded in the table when a match via `unqualified-search-registries` happen - is that correct?

Only in interactive use, i.e. not with CRI-O.

> From what I understood, `permissive` meant look at alias table, if no match is found don't prompt the user asking where to pull from, just check all the registries under `unqualified-search-registries`. And `disabled` meant don't look at any alias table, just check all the registries under `unqualified-search-registries`.

No, the mode _only_ governs lookup in USR (and possibly recording aliases for the outcome of that USR lookup).
- disabled = old risky USR lookup.
- enforcing = prompt interactively, fail if impossible.
- permissive = prompt if interactive, old risky USR lookup if not interactive.

> If there is truly no difference, then would it matter whether it is in `permissive` or `disabled` mode when it comes to alias look up?

No.